src/language/from_relay/mod.rs

```diff
@@ -655,29 +671,51 @@ fn compile_expression(
         .clone()
         .downcast::<tvm::ir::relay::TupleGetItem>()
     {
-        // We've only implemented tuple functionality for handling batch norms,
-        // at the moment. So we check to make sure this is handling a batch
-        // norm.
-
-        let arg = tuple_get_item
+        if let Ok(call) = tuple_get_item
             .tuple
             .clone()
             .downcast::<tvm::ir::relay::Call>()
-            .unwrap();
-        assert_eq!(
-            arg.op
-                .clone()
-                .upcast::<tvm::ir::expr::BaseExpr>()
-                .downcast::<tvm::ir::op::Op>()
-                .unwrap()
-                .name
-                .as_str(),
-            Ok("nn.batch_norm")
-        );
-        assert_eq!(tuple_get_item.index, 0);
+        {
+            // For batch norms, we use a hack where if
+            // simplify_batch_norm_for_inference_hack is set,
+            // we compile the Relay batch norm to a single output.
+
+            // All other expressions are compiled using Language::TupleGetItem construct
+
+            if simplify_batch_norm_for_inference_hack
+                && call
+                    .op
+                    .clone()
+                    .upcast::<tvm::ir::expr::BaseExpr>()
+                    .downcast::<tvm::ir::op::Op>()
+                    .unwrap()
+                    .name
+                    .as_str()
+                    .unwrap()
+                    == "nn.batch_norm"
+                && tuple_get_item.index == 0
+            {
+                get_compiled_expression(tuple_get_item.tuple.clone())
+            } else {
+                let data_id = get_compiled_expression(tuple_get_item.tuple.clone());
+                let index_id = glenside_expr.add(Language::Usize(tuple_get_item.index as usize));
+                glenside_expr.add(Language::TupleGetItem([data_id, index_id]))
+            }
+        } else {
+            let data_id = get_compiled_expression(tuple_get_item.tuple.clone());
+            let index_id = glenside_expr.add(Language::Usize(tuple_get_item.index as usize));
+            glenside_expr.add(Language::TupleGetItem([data_id, index_id]))
```

Comment:
This logic is duplicated. If you delete 700-702, I'm fairly certain this code will do exactly the same thing, because we'll just fall through to the outer `else` branch (the branch on 705-707). Can you check my thinking on that? I'd rather not duplicate logic if possible!

Reply:
hmm I gave it a try and RUst's type checker didn't like it. It looks its of the if expression, so the inner if-else expression needs type Egg::Id, and if the lines on 700-702 are deleted, Rust implicitly has a unit type

Reply:
I see. I changed it -- I just used a `return` to short-circuit.